more usecases (#17)
ElektraInitiative/PermaplanT: doc/usecases/label_layer.md

> # Use Case: Label Layer

Reviewer: ```suggestion
# Use Case: Labels Layer
```

## Summary

> - **Scope:** Label Layer

Reviewer: ```suggestion
- **Scope:** Labels Layer
```

- **Level:** User Goal
- **Actors:** App User

> - **Brief:** The user can add labels to their map to identify different elements or areas

Reviewer: ```suggestion
- **Brief:** The user adds labels to their map to identify different elements or areas
```

- **Status:** Draft

## Scenarios

- **Precondition:**

>   The user has opened the app and selected the label layer.

Reviewer: ```suggestion
  The user has opened the app and selected the labels layer.
```

- **Main success scenario:**

>   The user successfully adds labels to their map by selecting the element or area they want to label and entering the desired text in the label field.

Reviewer: ```suggestion
  The users successfully add labels to their map by selecting the element or area they want to label and entering the desired text in the label field.
```

  The label is displayed on the map and can be edited or removed at any time.
- **Alternative scenario:**
  The user accidentally adds a label in the wrong location and uses the app's undo function to correct the mistake
- **Error scenario:**

>   The user tries to add a label with more characters than the maximum allowed, and the app displays an error message indicating the maximum character limit.

Reviewer: Is the limit needed? How long will that limit be.

Author: I think a sensible limit is not bad since we can also prevent any possible attacks by users trying to overload the system with an infinite amount of strings.

My first thought is to use around 2,000 characters, that should be between 285 words and 500 words and I think that's plenty for labeling

Reviewer: Doesn't some token or similar (that only logged in users get) already protect against this? But as the labels are directly shown on the map, I also don't see any possibility that 2000 characters per label wouldn't be enough.

Nevertheless, I would prefer if the UI simply prevents to enter more than 2000 characters and the error message (needed on API level) actually should never be seen by the user.
